gapic/src/main/com/google/gapid/views/TracerDialog.java

```diff
@@ -597,20 +597,36 @@ private void updateDevicesDropDown(Settings settings) {
         if (device != null && devices != null) {
           deviceLoader.stopLoading();
           device.setInput(devices);
-          if (!settings.traceDevice.isEmpty()) {
-            Optional<DeviceCaptureInfo> deflt = devices.stream()
-                .filter(dev -> settings.traceDevice.equals(dev.device.getSerial()))
-                .findAny();
-            if (deflt.isPresent()) {
-              device.setSelection(new StructuredSelection(deflt.get()));
+          DeviceCaptureInfo deflt = getPreviouslySelectedDevice(settings).orElseGet(() -> {
+            if (devices.size() == 1) {
+              devices.get(0);
```

Comment:
I guess you meant: `return devices.get(0)`

Reply:
Done.